samples/peripheral/radio_test/src/radio_test.c

```diff
@@ -66,6 +68,50 @@ static uint8_t ppi_radio_start;
 static struct radio_test_fem fem;
 #endif /* CONFIG_FEM */
 
+static uint16_t channel_to_frequency(nrf_radio_mode_t mode, uint8_t channel)
+{
+#if CONFIG_HAS_HW_NRF_RADIO_IEEE802154
+	if (mode == NRF_RADIO_MODE_IEEE802154_250KBIT) {
+		if ((channel >= IEEE_MIN_CHANNEL) &&
+			(channel <= IEEE_MAX_CHANNEL)) {
+			return CHAN_TO_FREQ(IEEE_FREQ_CALC(channel));
+		} else {
+			return CHAN_TO_FREQ(IEEE_DEFAULT_FREQ);
+		}
+	} else {
+		return CHAN_TO_FREQ(channel);
+	}
+#else
+	return CHAN_TO_FREQ(channel);
+#endif /* CONFIG_HAS_HW_NRF_RADIO_IEEE802154 */
+}
+
+static void radio_power_set(nrf_radio_mode_t mode, uint8_t channel, int8_t power)
+{
+	int8_t output_power;
+
+#if CONFIG_FEM
+	uint16_t frequency;
+
+	if (IS_ENABLED(CONFIG_RADIO_TEST_POWER_CONTROL_AUTOMATIC)) {
+		frequency = channel_to_frequency(mode, channel);
+		output_power = fem_tx_output_power_set(power, frequency);
+	} else
+#else
+	ARG_UNUSED(mode);
+	ARG_UNUSED(channel);
+#endif /* CONFIG_FEM */
+	{
+		nrf_radio_txpower_set(NRF_RADIO, power);
+		output_power = power;
+	}
```

Comment:
```suggestion
	if (IS_ENABLED(CONFIG_FEM) && IS_ENABLED(CONFIG_RADIO_TEST_POWER_CONTROL_AUTOMATIC)) {
		uint16_t frequency;

		frequency = channel_to_frequency(mode, channel);
		output_power = fem_tx_output_power_set(power, frequency);
	} else {
		nrf_radio_txpower_set(NRF_RADIO, power);
		output_power = power;
	}
```

Reply:
Actually, since `RADIO_TEST_POWER_CONTROL_AUTOMATIC` depends on `FEM` the if condition above can be simplified:
```c
if (IS_ENABLED(CONFIG_RADIO_TEST_POWER_CONTROL_AUTOMATIC) {
   ...
}

Reply:
It would be nice but it is not possible, the function related to FEM are available only when `CONFIG_FEM` is enabled because we don't link FEM library and MPSL when this config is disabled. If I would use here `IS_ENABLE` macro I would get compilation warnings in best case.